### PR TITLE
Update to components 6.0.0 icon syntax

### PR DESCRIPTION
**Pull request by joebochill**
#### Changes proposed in this Pull Request:
- Updates to latest alpha version of 6.0.0 components
- Update the syntax used for icons
- 

#### To Test:
iOS
- clone
- yarn
- cd ios
- pod install
- cd .. && yarn ios

Android
- clone
- yarn
- yarn android

### CODE REVIEW

**Review by daileytj — APPROVED** [review]

File: App.tsx [hunk 1]
```diff
@@ -124,53 +110,53 @@ export const App: React.FC<AppProps> = ({ navigation }) => {
...
                             value={'5th Item'}
-                            IconClass={Battery}
+                            icon={Battery}
                             iconColor={theme.dark ? PXBColors.blue[200] : PXBColors.blue[500]}
                         />
                     </HeroBanner>
                     <InfoListItem
                         divider={'full'}
-                        IconClass={Sunny}
+                        icon={Sunny}
```
Comment (daileytj) [inline App.tsx:143]: Is the default color correct here?

File: App.tsx [hunk 2]
```diff
@@ -256,18 +242,18 @@ export const App: React.FC<AppProps> = ({ navigation }) => {
...
                         />
                         <InfoListItem
                             dense
-                            IconClass={Info}
+                            icon={Info}
                             iconColor={theme.dark ? PXBColors.blue[200] : PXBColors.blue[500]}
                             title={'1 Event'}
                         />
-                        <InfoListItem dense IconClass={Clouds} title={'Online'} />
+                        <InfoListItem dense icon={Clouds} title={'Online'} />
```
Comment (daileytj) [inline App.tsx:256]: Correct default color?

File: App.tsx [hunk 3]
```diff
@@ -293,14 +279,14 @@ export const App: React.FC<AppProps> = ({ navigation }) => {
                     ]}
                     badge={
                         <HeroBanner style={{ flex: 0, minWidth: 180, justifyContent: 'flex-end' }}>
-                            <Hero label={'Temperature'} iconSize={70} value={69} units={'°F'} IconClass={Temp} />
+                            <Hero label={'Temperature'} iconSize={70} value={69} units={'°F'} icon={Temp} />
```
Comment (daileytj) [inline App.tsx:282]: Correct default color?

**Comment by joebochill** [comment]
The default icon color for InfoListItem in Figma is gray[500]/black[200] (light/dark). Our RN implementation uses theme.colors.text which is black[500]/black[50]. This is a bug and should be logged on the RN components library (make sure to add a note to be careful about the items in the drawer, because they should use different defaults)

The hero defaults to theme.colors.text, which I believe is correct according to the docs (there are actually no examples of Hero in figma with the default color, but I assume matching the text is the expected default). EmptyState should use a different shade of gray, but Huayun has already logged a bug for that.

**Comment by joebochill** [comment]
See https://github.com/pxblue/react-native-component-library/issues/183

**Pull request merged**